### PR TITLE
Remove the need to wrap tests

### PR DESCRIPTION
**Pull request by fermayo**

### CODE REVIEW

File: testing.go [hunk 1]
```diff
@@ -8,7 +8,19 @@ import (
...
 
+type Test struct {
+	ctx  context.Context
+	span opentracing.Span
+	t    *testing.T
+}
+
 func InstrumentTest(t *testing.T, f func(ctx context.Context, t *testing.T)) {
+	test := StartTest(t)
+	defer test.End()
```
Comment (tonyredondo) [inline testing.go]: I thought you were going to return both `test` and `ctx` as a separated variables. But I guess is cleaner this way.
Also the `InstrumentTest` function kept the `func` as parameter?

File: testing.go [hunk 2]
```diff
@@ -8,7 +8,19 @@ import (
...
 )
 
+type Test struct {
+	ctx  context.Context
+	span opentracing.Span
+	t    *testing.T
+}
+
 func InstrumentTest(t *testing.T, f func(ctx context.Context, t *testing.T)) {
+	test := StartTest(t)
```
Comment (tonyredondo) [inline testing.go]: I thought you were going to return both `test` and `ctx` as a separated variables. But I guess is cleaner this way.
Also you are keeping both the pair `StartTest` - `End` and `InstrumentTest` way, Is there a reason to no deprecate the last one?

Comment (fermayo) [reply testing.go]: It was easy to still support it

Comment (fermayo) [reply testing.go]: > I thought you were going to return both test and ctx as a separated variables. But I guess is cleaner this way.

I was doing that at the beginning, but it's more verbose for tests that don't use `ctx`. You can still use `test.Context()` which seems to be the standard way to get context objects

**Pull request merged**